### DISCUSSION TITLE
Training a custom NER model using a pre-existing model (en_core_web_sm)

### DISCUSSION THREAD

**Discussion by Shakeeey** [explosion/spaCy#12719] [Help: Model Advice]
I have a custom NER model that is created using train and test .spacy files. I have watched many tutorials and they all use spacy.blank("en") but I want to use the "en_core_web_sm" model and append my own custom entities to it. I have watched videos on how to do it on spacy v2 but nothing on v3 yet.

I have tried changing the config file and updating the [components.ner] to source = "en_core_web_sm" but I still don't get the pre-existing entities like (GPE, Date, Person, etc.) in my output. 

Here is the link to the same question with maybe more details in a stackoverflow question: https://stackoverflow.com/questions/76459442/spacy-v3-custom-ner-model-training

**Comment by kadarakos** [comment]
Hey Shakeeey,

To be able to give you better help may I ask you to post the config files that you are using? You can also take a look at a related thread, not exactly your issue, but it might give you some useful information: https://github.com/explosion/spaCy/discussions/10674#discussion-4020374

**Reply by Shakeeey** [reply]
Hey,

here is the link of my repo for this project: https://github.com/Shakeeey/spacy_project

the double.cfg is for combining those two models and the config.cfg is for creating my custom model. 

Note: I used this tutorial to follow along to merge them together - https://github.com/explosion/projects/tree/v3/tutorials/ner_double

**Reply by kadarakos** [reply]
It seems like you are sourcing the `ner` component from the `en_core_web_sm`: https://github.com/Shakeeey/spacy_project/blob/main/config.cfg#L24, however, its corresponding `tok2vec` does not seem to be sourced: https://spacy.io/usage/processing-pipelines#sourced-components. 

In general we do not recommend starting from a pre-trained `ner` model because its easy for it to forget what it learned when it sees new data. I recommend just training a fresh `ner` component, but using the `en_core_web_lg` vectors or if you have a enough resourced using a `transformer` backbone.

**Reply by kadarakos** [reply]
Oh sorry, I made a mistake the `ner` component actually has it's own `tok2vec`! Nevertheless, we do not recommend fine-tuning it.
